File: php/tests/searchsettings_test.php
<?php

class SearchSettingsTest extends PHPUnit_Framework_TestCase {
    function __construct() {
        parent::__construct();
        $this->settings = new SearchSettings();
    }

    public function test_default_settings() {
        $this->assertFalse($this->settings->archivesonly);
        $this->assertFalse($this->settings->debug);
        $this->assertFalse($this->settings->dotiming);
        $this->assertTrue($this->settings->excludehidden);
        $this->assertFalse($this->settings->firstmatch);
        $this->assertEquals($this->settings->linesafter, 0);
        $this->assertEquals($this->settings->linesbefore, 0);
        $this->assertFalse($this->settings->listdirs);
        $this->assertFalse($this->settings->listfiles);
        $this->assertFalse($this->settings->listlines);
        $this->assertEquals($this->settings->maxlinelength, 150);
        $this->assertFalse($this->settings->multilinesearch);
        $this->assertTrue($this->settings->printresults);
        $this->assertFalse($this->settings->printusage);
        $this->assertFalse($this->settings->printversion);
        $this->assertTrue($this->settings->recursive);
        $this->assertFalse($this->settings->searcharchives);
        $this->assertFalse(isset($this->settings->startpath));
        $this->assertFalse($this->settings->uniquelines);
        $this->assertFalse($this->settings->verbose);
    }

    public function test_set_properties() {
        $this->settings->archivesonly = true;
        $this->settings->debug = true;
        $this->settings->dotiming = true;
        $this->settings->linesafter = 5;
        $this->settings->linesbefore = 5;
        $this->assertTrue($this->settings->archivesonly);
        $this->assertTrue($this->settings->debug);
        $this->assertTrue($this->settings->dotiming);
        $this->assertEquals($this->settings->linesafter, 5);
        $this->assertEquals($this->settings->linesbefore, 5);
    }

    public function test_add_single_extension() {
        $this->settings->add_exts('php', $this->settings->in_extensions);
        $this->assertEquals(count($this->settings->in_extensions), 1);
        $this->assertEquals($this->settings->in_extensions[0], 'php');
    }

    public function test_add_comma_delimited_extensions() {
        $this->settings->add_exts('php,py', $this->settings->in_extensions);
        $this->assertEquals(count($this->settings->in_extensions), 2);
        $this->assertTrue(in_array('php', $this->settings->in_extensions));
        $this->assertTrue(in_array('py', $this->settings->in_extensions));
    }
}

?>

File: ruby/tests/searchsettings_test.rb
################################################################################
#
# filetypes_test.rb
#
# Test the FileTypes class
#
################################################################################

require_relative "../rbsearch/searchsettings.rb"
require "test/unit"
 
class SearchSettingsTest < Test::Unit::TestCase
  def setup
    @settings = SearchSettings.new
  end

  def test_default_settings
    assert_equal(@settings.archivesonly, false)
    assert_equal(@settings.debug, false)
    assert_equal(@settings.excludehidden, true)
    assert_equal(@settings.firstmatch, false)
    assert_equal(@settings.linesafter, 0)
    assert_equal(@settings.linesbefore, 0)
    assert_equal(@settings.listdirs, false)
    assert_equal(@settings.listfiles, false)
    assert_equal(@settings.listlines, false)
    assert_equal(@settings.maxlinelength, 150)
    assert_equal(@settings.multilinesearch, false)
    assert_equal(@settings.printresults, true)
    assert_equal(@settings.printusage, false)
    assert_equal(@settings.printversion, false)
    assert_equal(@settings.recursive, true)
    assert_equal(@settings.searcharchives, false)
    assert_equal(@settings.startpath, nil)
    assert_equal(@settings.uniquelines, false)
    assert_equal(@settings.verbose, false)
    assert(@settings.in_archiveextensions.empty?)
    assert(@settings.in_archivefilepatterns.empty?)
    assert(@settings.in_dirpatterns.empty?)
    assert(@settings.in_filepatterns.empty?)
    assert(@settings.in_linesafterpatterns.empty?)
    assert(@settings.in_linesbeforepatterns.empty?)
    assert(@settings.linesaftertopatterns.empty?)
    assert(@settings.linesafteruntilpatterns.empty?)
    assert(@settings.out_archiveextensions.empty?)
    assert(@settings.out_archivefilepatterns.empty?)
    assert(@settings.out_dirpatterns.empty?)
    assert(@settings.out_filepatterns.empty?)
    assert(@settings.out_linesafterpatterns.empty?)
    assert(@settings.out_linesbeforepatterns.empty?)
    assert(@settings.searchpatterns.empty?)
  end

  def test_set_properties
    @settings.archivesonly = true
    @settings.debug = true
    @settings.linesafter = 5
    @settings.linesbefore = 5
    assert_equal(@settings.archivesonly, true)
    assert_equal(@settings.debug, true)
    assert_equal(@settings.linesafter, 5)
    assert_equal(@settings.linesbefore, 5)
  end

  def test_add_single_extension
    @settings.add_exts('rb', @settings.in_extensions)
    self.assert_equal(@settings.in_extensions.length, 1)
    self.assert(@settings.in_extensions.include?('rb'))
  end

  def test_add_comma_delimited_extensions
    @settings.add_exts('py,rb', @settings.in_extensions)
    assert_equal(@settings.in_extensions.length, 2)
    assert(@settings.in_extensions.include?('py'))
    assert(@settings.in_extensions.include?('rb'))
  end

  def test_add_pattern
    @settings.add_pattern('Search', @settings.searchpatterns)
    assert_equal(@settings.searchpatterns.length, 1)
    assert_equal(@settings.searchpatterns.first.source, 'Search')
  end
end
